gate-web/src/main/groovy/com/netflix/spinnaker/gate/services/PipelineService.groovy

```diff
@@ -96,6 +96,26 @@ class PipelineService {
     orcaServiceSelector.withContext(RequestContext.get()).startPipeline(pipelineConfig, trigger.user?.toString())
   }
 
+  Map triggerViaEcho(String application, String pipelineNameOrId, Map parameters) {
```

Comment:
What are the security controls around this?

With respect to pipeline triggering, I don't think _echo_ is doing any validation and will just blindly execute as the service account.

Reply:
Good point about security; I actually am not overly familiar with how our auth works.  Travis gave me an overview yesterday and I'm going to see if/how I can keep the same auth controls when going through echo.